docs/adr/adr_backend_etl_and_hosting_for_StatViz.md

```diff
@@ -0,0 +1,237 @@
+
+#### A.1 About OLAP Cubes and Star Schema:
+
+Our inspiration comes from the architectures of data warehouses, especially OLAP cubes, which provide insights through data analysis.
+
+We intend to standardize the return data using a star schema to facilitate the creation of reusable components. A sample structure of the returned GraphQL query is:
+
+```
+dataQueryForVisualization {
+  facts: [{
+    dim1_Id: ...,
+    dim2_Id: ...,
+    fact: ...
+  }, ...],
+  dimensions: {
+    dim1: [{id: 1, name: ....}, ...],
+    dim2: [...]
+  }
+}
+```
+
+#### A.2 Determining Visualization Dimensions:
+
+The hard part is to determine the best set of returned dimensions for each visualization. In general, we can differentiate the dimensions into three categories:
+
+1. **Basic Dimesnions**: At the very least, the returned dimensions should be the foundation of the visualization. For instance, a demographic plot requires dimensions like gender and age.
+2. **User Filters**: If users can apply filters, such as a creation date, including these dimensions lets the frontend dynamically adjust the data. Excluding these necessitates backend-side data preparation and passing them as GraphQL query variables.
+3. **URL Filters**: Filters, like which organization or base a visualization targets, can be passed as dimensions or as GraphQL query variables.
+
+#### A.3 N-to-N dimensions
+
+Handling many-to-many (N-to-N) relationships in OLAP cubes can be tricky, as they can introduce ambiguity and potential double counting of facts.
+
+For instance, while examining beneficiary demographics, a potential N-to-N relationship is the "tags". We considered a star scheme that looks like:
+
+```
+beneficiaryDemographics {
+  facts: [{
+    gender: String,
+    age: Int,
+    tagId: Int,
+    count: Int
+  }, ...],
+  dimensions: {
+    tag: [{id: Int, name: String}, ...],
+  }
+}
+```
+
+However, grouping by gender or age can cause double counting due to lost relational information among tags.
+
+To tackle this, we chose a non-standard star scheme. Since our data isn't restricted to relational table layouts, dimensions can be represented by arrays. This leads to:
+
+```
+beneficiaryDemographics {
+  facts: [{
+    gender: String,
+    age: Int,
+    tagIds: Int[],
+    count: Int
+  }, ...],
+  dimensions: {
+    tag: [{id: 1, name: ....}, ...],
+  }
+}
+```
+
+### B. Data Transformation
+
+The BE has to transform Boxtribute's data to be able to return the data in the desired fromat from above. Now, the question arises when and how to do these transformations. We see the following options:
+
+#### B.1 On-the-fly Transformation During GraphQL Requests
+
+This is straight-forward solution and how we started for the first visualizations - Directly transform operational data from Boxtribute in response to incoming graphQL requests.
+
+**Pros**:
+
+- you only need to write the query
+- live data
+
+**Cons**:
+
+- query with many reads directly from the Boxtribute db
```

Comment:
another con (maybe you see it implicitly covered): historically accurate data needs to be reconstructed